src/main/java/com/microsoft/jenkins/acs/commands/GetContainerServiceInfoCommand.java

```diff
@@ -36,21 +35,19 @@ public void execute(IGetContainerServiceInfoCommandData context) {
         JobContext jobContext = context.getJobContext();
         final FilePath workspace = jobContext.getWorkspace();
         final TaskListener taskListener = jobContext.getTaskListener();
-        String azureCredentialsId = context.getAzureCredentialsId();
-        final AzureCredentials.ServicePrincipal servicePrincipal =
-                AzureCredentials.getServicePrincipal(azureCredentialsId);
+        final String azureCredentialsId = context.getAzureCredentialsId();
         final String resourceGroupName = context.getResourceGroupName();
         final String containerServiceName = context.getContainerServiceName();
         final String containerServiceType = context.getContainerServiceType();
-        final ContainerServiceOchestratorTypes configuredType = context.getOrchestratorType();
+        final ContainerServiceOrchestratorTypes configuredType = context.getOrchestratorType();
 
         final String aiType = AzureACSPlugin.normalizeContainerSerivceType(containerServiceType);
 
+        Azure azureClient = AzureHelper.buildClient(azureCredentialsId);
         AzureACSPlugin.sendEventFor(Constants.AI_START_DEPLOY,
                 aiType,
                 jobContext.getRun(),
-                "Subscription", AppInsightsUtils.hash(
-                        servicePrincipal == null ? null : servicePrincipal.getSubscriptionId()),
+                "Subscription", AppInsightsUtils.hash(azureClient.subscriptionId()),
```

Comment:
what if the provided credentials are invalid, will the build job terminated in exception rather than logging in AI for the failure and then fails?